### PR TITLE
✨ style: Update Messages Style to Match ChatGPT (ChatGPT UI)

### PR DESCRIPTION
**Pull request by techwithanirudh**
## Summary

Modernized the message components to match ChatGPT’s style and optimized dark mode support.
> Note: ⚠️ This P.R. is one of multiple smaller P.R.s split from a larger one by GPT-4o. 

### Desktop
Before:
image

After:
image

Before:
image

After:
image

### Mobile
Before:
image

After:
image

Before:
image

After:
image

## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Testing

N/A

### **Test Configuration**:

N/A

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes

### CODE REVIEW

**Comment by techwithanirudh** [comment]
@danny-avila Could you please review this P.R?

**Comment by berry-13** [comment]
@techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?

**Comment by techwithanirudh** [comment]
> @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?

Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

**Comment by berry-13** [comment]
> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

thanks!

**Comment by danny-avila** [comment]
> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.



> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.



> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

I won't be merging this PR unless there is a way to select different themes for the new UI updates, with the current theme being the default.

**Comment by techwithanirudh** [comment]
@danny-avila This has already been implemented in this P.R

**Comment by danny-avila** [comment]
> @danny-avila This has already been implemented in this P.R, if this P.R gets merged I could update the others to do the same.

Ok, I didn't realize that. I will try it out.

**Comment by techwithanirudh** [comment]
> > @danny-avila This has already been implemented in this P.R, if this P.R gets merged I could update the others to do the same.
> 
> Ok, I didn't realize that. I will try it out.

@danny-avila Thanks! I'll resolve the merge conflicts

File: client/src/store/settings.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ const staticAtoms = {
 // Atoms with localStorage
 const localStorageAtoms = {
   autoScroll: atomWithLocalStorage('autoScroll', false),
+  messagesUI: atomWithLocalStorage('messagesUI', true),
```
Comment (danny-avila) [inline client/src/store/settings.ts]: default value should be `false`

File: client/src/localization/languages/Eng.ts [hunk 2]
```diff
@@ -479,6 +479,7 @@ export default {
   com_nav_welcome_assistant: 'Please Select an Assistant',
   com_nav_welcome_message: 'How can I help you today?',
   com_nav_auto_scroll: 'Auto-Scroll to latest message on chat open',
+  com_nav_messages_ui: 'Enable new UI',
```
Comment (danny-avila) [inline client/src/localization/languages/Eng.ts]: "Use Bubble styling for User Messages"

File: client/src/components/Nav/SettingsTabs/General/General.tsx [hunk 3]
```diff
@@ -163,6 +164,9 @@ function General() {
         <div className="border-b pb-3 last-of-type:border-b-0 dark:border-gray-600">
           <AutoScrollSwitch />
         </div>
+        <div className="border-b pb-3 last-of-type:border-b-0 dark:border-gray-600">
```
Comment (danny-avila) [inline client/src/components/Nav/SettingsTabs/General/General.tsx]: This should go in a new Tab called "Appearance"

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by danny-avila** [comment]
The hover buttons should also hover when the entire message row is being hovered in the new style. currently it does not, and it doesn't feel intuitive to hover over the bubble first.


https://github.com/danny-avila/LibreChat/assets/110412045/52b0ac5f-9f86-45e3-a215-9ca2f2e66728

**Comment by techwithanirudh** [comment]
> The hover buttons should also hover when the entire message row is being hovered in the new style. currently it does not, and it doesn't feel intuitive to hover over the bubble first.
> 
>  Recording.2024-06-08.183608.mp4

@danny-avila I tried to fix this issue, is this fix okay?

**Comment by danny-avila** [comment]
This PR was closed due to fundamental changes made to messages to accommodate this PR https://github.com/danny-avila/LibreChat/pull/3191

I'm still open to the bubble message look as an option but with geared for dependency injection from the start, with minimal code edits to the main message components and Hover buttons.

I closed this PR as I don't expect the changes to be reconciled from the current state of the PR which already had some merge conflicts before #3191

**Pull request closed without merging**